### ISSUE TITLE
Be randomized, avoid _seed

### ISSUE THREAD

**Issue by breznak** [numenta/nupic-legacy#433]
In a lot of places in the code, we set random seed to obtain reproducible results. 
While it's perfectly correct to use for bit-to-bit comparisons of output - eg cpp/py implementation of SP, its use in examples to obtain "good" values for datasets is wrong. 

It's just overfitting the parameters on the dataset when results with different random would drop rapidly in quality. 

Another good use is randomization is it's dodgy, we'll have more problems, but find and fix bugs faster.

**Comment by rhyolight** [comment]
@subutai said:
> First, the new TM code should be changed to use the RNG in nupic.core. I didn't realize it wasn't using it.

[It looks like this has happened now](https://github.com/numenta/nupic/blob/master/src/nupic/research/temporal_memory.py#L29).

**Comment by rhyolight** [comment]
### Please review this issue

This issue needs to be reviewed by the original author or another contributor for applicability to the current codebase. The issue might be obsolete or need updating to match current standards and practices. If the issue is out of date, please close. Otherwise please leave a comment to justify its continuing existence. It may be closed in the future if no further activity is noted.

* * *

I marked this as under review because IMO the scope is too big. It needs to be broken up into smaller tasks after a thorough review of the current codebase of the use of random numbers.

**Comment by cogmission** [comment]
> At that point all we would need is a mock API for other languages such as the Java port. We can have functionality in core that outputs the random numbers for a given seed to a file so that the Java mock API can read it in and replicate the core results. (Unless I'm missing something, I don't think our python or c++ code needs to call the mock API, since they can just use the real API and get exactly reproducible results.) 

I agree. I believe this additional part should be added as another issue having to do with enabling the consistency of RNG output across languages.
